modules/core/src/command_line_parser.cpp

```diff
@@ -134,22 +135,25 @@ void CommandLineParser::getByIndex(int index, bool space_delete, int type, void*
                 String v = impl->data[i].def_value;
                 if (space_delete == true) v = impl->cat_string(v);
 
-                // it is an error if we just got the default value here
-                if(v.empty())
-                    break;
-
+                // the key was neither specified nor has it a default value
+                if(v.empty()) {
```

Comment:
Actually, I mean this fix for `.getByName()` method at first (which breaks existed perf tests). But `.getByIndex()` should be fixed also.

In my sample above, the empty string is not an error, so I don't need error handling from `CommandLineParser` in this case.

> Then there is an inconsistency for positional arguments between calling get<String>(0) and get<String>("@foo") and between positional and keyword arguments in general.

`.getByName()` condition is fixed, so lets fix `.getByIndex()` condition and allow empty strings (with fix for check `parser.check() == true` in `testPositional_noArgs`)

> Would you agree with adding an operation mode where the parser disallows empty default strings?

I suggest to make an another PR with these changes. This PR is good enough.